shared-helpers/src/views/summaryTables.tsx

```diff
@@ -355,6 +441,167 @@ export const stackedUnitSummariesTable = (
   return [rowData]
 }
 
+export const getAvailabilityText = (
+  group: UnitGroupSummary,
+  isComingSoon?: boolean
+): { text: string; subText: string } => {
+  const hasVacantUnits = group.unitVacancies > 0
+  const waitlistStatus = group.openWaitlist
+    ? t("listings.waitlist.open")
+    : t("listings.waitlist.closed")
+
+  // Create an array of status elements to combine
+  const statusElements = []
+
+  if (group.openWaitlist) {
+    statusElements.push(waitlistStatus)
+  }
+
+  if (isComingSoon) {
+    statusElements.push(t("listings.underConstruction"))
+  }
+
+  // Combine statuses with proper formatting
+  let availability = null
+  if (statusElements.length > 0) {
+    if (statusElements.length >= 3) {
+      const lastElement = statusElements.pop()
+      availability = `${statusElements.join(", ")} & ${lastElement}`
+    } else {
+      availability = statusElements.join(" & ")
+    }
+  }
+
+  const text = availability ?? t("listings.unitsSummary.notAvailable")
+  const subText = hasVacantUnits
+    ? `${group.unitVacancies} ${
+        group.unitVacancies === 1 ? t("listings.vacantUnit") : t("listings.vacantUnits")
+      }`
+    : ""
+
+  return { text, subText }
+}
+
+export const getAvailabilityTextForGroup = (
+  groups: UnitGroupSummary[],
+  isComingSoon?: boolean
+): { text: string; subText: string } => {
+  // Track all statuses across groups
+  const statusSet = new Set<string>()
+  let totalVacantUnits = 0
+
+  // Collect information from all groups
+  groups.forEach((group) => {
+    if (group.openWaitlist) {
+      statusSet.add(t("listings.waitlist.open"))
+    }
+
+    if (group.unitVacancies > 0) {
+      totalVacantUnits += group.unitVacancies
+    }
+  })
+
+  // Add coming soon status if needed
+  if (isComingSoon) {
```

Comment:
question: Should `isComingSoon` replace the whole text with "Under Construction" even if there are "available units"? It seems weird to display both  "open waitlist" and "Under Construction"
![image](https://github.com/user-attachments/assets/4ecafae2-a2f0-44ba-a83a-d144f012f0de)

Reply:
@KrissDrawing I got confirmation from Sarah that if the listing is marked as "Under Construction" the "Availability" column should only show that even if the group has been marked as open waitlist. Also the "5 Vacant Units" should not show in this case as well.